### PR TITLE
Store user data in memory on login for stateless auth adapters

### PR DESCRIPTION
**Pull request by biesbjerg**
Ref #9070

### CODE REVIEW

**Comment by markstory** [comment]
To be that guy, is there a way we could get tests?

**Comment by biesbjerg** [comment]
Sure you can! Test added.

image

**Comment by markstory** [comment]
Looks good to me.

File: lib/Cake/Test/Case/Controller/Component/AuthComponentTest.php [hunk 1]
```diff
@@ -1722,6 +1722,25 @@ public function testStatelessAuthNoRedirect() {
...
+			'id' => 1,
+			'username' => 'mark'
+		);
+
+		AuthComponent::$sessionKey = false;
+		$result = $this->Auth->login($user);
+		$this->assertTrue($result);
+
+		$this->assertTrue($this->Auth->loggedIn());
+		$this->assertEquals($user, $this->Auth->user());
```
Comment (ADmad) [inline lib/Cake/Test/Case/Controller/Component/AuthComponentTest.php]: Please also add a `$this->assertFalse($this->Auth->Session->started());`

**Comment by biesbjerg** [comment]
Good call. Added an test is passing.

This is my first time doing testing. All Auth session tests fail when running from the CLI - is that expected?

File: lib/Cake/Test/Case/Controller/Component/AuthComponentTest.php [hunk 2]
```diff
@@ -1722,6 +1722,27 @@ public function testStatelessAuthNoRedirect() {
...
+
+		$this->assertNull($this->Auth->Session->started());
```
Comment (ADmad) [inline lib/Cake/Test/Case/Controller/Component/AuthComponentTest.php]: `started()` returns boolean so this asserting will always fail :)

**Comment by biesbjerg** [comment]
Doh.

**Comment by ADmad** [comment]
> All Auth session tests fail when running from the CLI - is that expected?

No. Did you use the `--stderr` switch for phpunit?  You can check `.travis.yml` for the command used.

The patch and tests look good now.

**Comment by dereuromark** [comment]
:+1:

**Pull request merged**